app/controllers/courses_controller.rb

```diff
@@ -58,7 +58,7 @@ def statistics
       if current_user.id == @teacher_id
         @teacher = true
       elsif current_user.student?
-        @student = @terms.first.students.find_by user_id: current_user.id
+        @student = @terms.collect { |term| term.students.where(user_id: current_user.id) }.any?
```

Comment:
Because we want to select a student with certain id and certain course, but the course cannot be accessed straight from `Student` model, it's only possible for terms.